src/gpodder/plugins/soundcloud.py

```diff
@@ -38,12 +38,39 @@
 
 
 # gPodder's consumer key for the Soundcloud API
-CONSUMER_KEY = 'zrweghtEtnZLpXf3mlm8mQ'
+CLIENT_ID = "foo"
+CLIENT_SECRET = "bar"
 
+access_token = None
+token_expire = datetime.now()
 
 logger = logging.getLogger(__name__)
 
 
+class SoundcloudException(Exception):
+    pass
+
+
+def urlopen_with_token(url):
+    global access_token
+    global token_expire
+    if access_token is None or token_expire >= datetime.now():
+        response = util.urlopen("https://api.soundcloud.com/oauth2/token", method="POST", data={
+            "grant_type": "client_credentials",
+            "client_id": CLIENT_ID,
+            "client_secret": CLIENT_SECRET,
+        })
+        if response.status_code != 200:
+            raise SoundcloudException("Authentication not possible")
+        content = response.json()
+        access_token = content["access_token"]
+        token_expire = datetime.now() + timedelta(seconds=content["expires_in"] - 30)
```

Comment:
Here I'm decreasing the grace period by 30 seconds (It's 1h by default) to somewhat reduce racyness here and not make requests with expired tokens ever. I also thought of unsetting the access token and recursively calling this function, but that might have the potential for infinite recursion if something goes wrong